metrics/client_test.go

```diff
@@ -290,6 +291,72 @@ func TestTokenAuthenticationWithSSL(t *testing.T) {
 	assert.Equal(t, fmt.Sprintf("Bearer %s", p.Token), r.Header.Get("X-Authorization"))
 }
 
+func TestBasicAuthenticationWithSSL(t *testing.T) {
+	s := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("X-Authorization", r.Header.Get("Authorization"))
+	}))
+	defer s.Close()
+
+	tC := &tls.Config{InsecureSkipVerify: true}
+
+	p := Parameters{
+		Tenant:    "some tenant",
+		Url:       s.URL,
+		Username:  "user",
+		Password:  "pass",
+		TLSConfig: tC,
+	}
+
+	c, err := NewHawkularClient(p)
+	assert.NoError(t, err)
+
+	r, err := c.Send(c.URL("GET"))
+	assert.NoError(t, err)
+	assert.Equal(t, fmt.Sprintf("Basic %s", base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%v:%v", p.Username, p.Password)))), r.Header.Get("X-Authorization"))
+}
+
+func TestInvalidBasicAuthentication(t *testing.T) {
+	tC := &tls.Config{InsecureSkipVerify: true}
+
+	p := Parameters{
+		Tenant:    "some tenant",
+		Url:       "http://localhost:8080",
+		Username:  "user",
+		Password:  "pass",
+		Token:     "token",
+		TLSConfig: tC,
+	}
+
+	_, err := NewHawkularClient(p)
+	if err == nil {
+		t.Fatalf("Should not be able to specify both credentials and token")
+	}
+
+	p = Parameters{
+		Tenant:    "some tenant",
+		Url:       "http://localhost:8080",
+		Username:  "user",
+		TLSConfig: tC,
+	}
+
+	_, err = NewHawkularClient(p)
+	if err == nil {
```

Comment:
``assert.Error(t, err)`` would be more consistent, since everything else uses an error. Or even ``assert.ErrorEquals`` if you want to check the returned error is correct.